File: presentation/src/main/java/info/firozansari/presentation/internal/di/components/ApplicationComponent.java

package info.firozansari.presentation.internal.di.components;

import android.content.Context;
import info.firozansari.domain.executor.PostExecutionThread;
import info.firozansari.domain.executor.ThreadExecutor;
import info.firozansari.domain.repository.UserRepository;
import info.firozansari.presentation.internal.di.modules.ApplicationModule;
import info.firozansari.presentation.view.activity.BaseActivity;
import dagger.Component;
import javax.inject.Singleton;

/**
 * A component whose lifetime is the life of the application.
 */
@Singleton // Constraints this component to one-per-application or unscoped bindings.
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {
  void inject(BaseActivity baseActivity);

  //Exposed to sub-graphs.
  Context context();
  ThreadExecutor threadExecutor();
  PostExecutionThread postExecutionThread();
  UserRepository userRepository();
}

File: presentation/src/main/java/info/firozansari/presentation/internal/di/components/ApplicationComponent.kt
package info.firozansari.presentation.internal.di.components

import android.content.Context
import dagger.Component
import info.firozansari.domain.executor.PostExecutionThread
import info.firozansari.domain.executor.ThreadExecutor
import info.firozansari.domain.repository.UserRepository
import info.firozansari.presentation.internal.di.modules.ApplicationModule
import info.firozansari.presentation.view.activity.BaseActivity

/**
 * A component whose lifetime is the life of the application.
 */
// Constraints this component to one-per-application or unscoped bindings.
@Component(modules = [ApplicationModule::class])
interface ApplicationComponent {
    open fun inject(baseActivity: BaseActivity?)

    //Exposed to sub-graphs.
    open fun context(): Context?
    open fun threadExecutor(): ThreadExecutor?
    open fun postExecutionThread(): PostExecutionThread?
    open fun userRepository(): UserRepository?
}